File: server/routes/instances.js
// routes/instances.js
const express = require('express');
const router = express.Router();
const instancesController = require('../controllers/instancesController');

// GET /instances - List all active instances
router.get('/', async (req, res) => {
  try {
    const instances = await instancesController.listInstances();
    res.json(instances);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// GET /instances/details - List instances with detailed information
router.get('/details', async (req, res) => {
  try {
    const instances = await instancesController.listInstancesWithDetails();
    res.json(instances);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// POST /instances - Create a new instance for a test
router.post('/', async (req, res) => {
  try {
    const instanceData = req.body;
    const result = await instancesController.createInstance(instanceData);
    res.status(201).json(result);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// POST /instances/:id/report - Submit a report for an instance
router.post('/:id/report', async (req, res) => {
  try {
    const { id } = req.params;
    const text = req.body;
    
    if (!text) {
      return res.status(400).json({ error: 'Report content is required' });
    }

    const result = await instancesController.saveReport(id, text);
    res.status(201).json(result);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// GET /instances/:id/report - Get the last report for an instance
router.get('/:id/report', async (req, res) => {
  try {
    const { id } = req.params;
    const result = await instancesController.getLastReport(id);
    res.json(result);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// DELETE /instances/:id - Delete an instance
router.delete('/:id', async (req, res) => {
  try {
    const { id } = req.params;
    const result = await instancesController.deleteInstance(id);
    res.json(result);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

module.exports = router; 

File: server-python/routes/instances.py
from flask import Blueprint, request, jsonify
from controllers.instances_controller import create_instance, list_instances, delete_instance

instances_bp = Blueprint('instances', __name__)

@instances_bp.route('', methods=['GET'])
def get_instances():
    """Get all active instances"""
    try:
        instances = list_instances()
        return jsonify(instances)
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@instances_bp.route('', methods=['POST'])
def create_new_instance():
    """Create a new instance"""
    try:
        data = request.json
        
        # Extract parameters
        instance_name = data.get('instanceName')
        github_repo = data.get('githubRepo')
        github_token = data.get('githubToken')
        port_mapping = data.get('portMapping')
        initial_prompt = data.get('initialPrompt')
        final_prompt = data.get('finalPrompt')
        assessment_prompt = data.get('assessmentPrompt')
        
        if not instance_name:
            return jsonify({'error': 'Instance name is required'}), 400
        
        # Create instance
        instance = create_instance(
            instance_name=instance_name,
            github_repo=github_repo,
            github_token=github_token,
            port_mapping=port_mapping,
            initial_prompt=initial_prompt,
            final_prompt=final_prompt,
            assessment_prompt=assessment_prompt
        )
        
        return jsonify(instance), 201
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@instances_bp.route('/<instance_id>', methods=['DELETE'])
def delete_existing_instance(instance_id):
    """Delete an instance by ID"""
    try:
        result = delete_instance(instance_id)
        return jsonify(result)
    except Exception as e:
        return jsonify({'error': str(e)}), 500 